Sources/StreamChatUI/CommonViews/AvatarView/ChatChannelAvatarView.swift

```diff
@@ -30,31 +47,293 @@ open class ChatChannelAvatarView: _View, ThemeProvider, SwiftUIRepresentable {
-                .first(where: { $0.id != content.currentUserId })
-
-            return (firstOtherMember?.imageURL, firstOtherMember?.isOnline ?? false)
-        }()
-
+        loadAvatar(for: channel)
+    }
+    
+    open func loadAvatar(for channel: ChatChannel) {
+        // If the channel has an avatar set, load that avatar
+        if let channelAvatarUrl = channel.imageURL {
+            loadChannelAvatar(from: channelAvatarUrl)
+            return
+        }
+      
+        // Use the appropriate method to load avatar based on channel type
+        if channel.isDirectMessageChannel {
+            loadDirectMessageChannelAvatar(channel: channel)
+        } else {
+            loadMergedAvatars(channel: channel)
+        }
+    }
+    
+    /// Loads the avatar from the URL. This function is used when the channel has a non-nil `imageURL`
+    /// - Parameter url: The `imageURL` of the channel
+    open func loadChannelAvatar(from url: URL) {
         presenceAvatarView.avatarView.imageView.loadImage(
-            from: avatarURL,
+            from: url,
             placeholder: appearance.images.userAvatarPlaceholder4,
             preferredSize: .avatarThumbnailSize,
             components: components
         )
+    }
+    
+    /// Loads avatar for a directMessageChannel
+    /// - Parameter channel: The channel
+    open func loadDirectMessageChannelAvatar(channel: ChatChannel) {
+        let lastActiveMembers = channel.lastActiveMembers
+            .sorted { $0.memberCreatedAt < $1.memberCreatedAt }
+            .filter { $0.id != content.currentUserId }
+        
+        // If there are no members other than the current user in the channel, load a placeholder
+        if lastActiveMembers.isEmpty {
+            presenceAvatarView.avatarView.imageView.loadImage(
+                from: nil,
+                placeholder: appearance.images.userAvatarPlaceholder4,
+                preferredSize: .avatarThumbnailSize,
+                components: components
+            )
+            return
+        }
+        
+        let otherMember = lastActiveMembers.first
+        let isOnlineIndicatorVisible = otherMember?.isOnline ?? false
+        
+        presenceAvatarView.avatarView.imageView.loadImage(
+            from: otherMember?.imageURL,
+            placeholder: appearance.images.userAvatarPlaceholder3,
+            preferredSize: .avatarThumbnailSize,
+            components: components
+        )
+        
         presenceAvatarView.isOnlineIndicatorVisible = isOnlineIndicatorVisible
     }
+    
+    /// Loads an avatar which is merged (tiled) version of the first four active members of the channel
+    /// - Parameter channel: The channel
+    open func loadMergedAvatars(channel: ChatChannel) {
+        // The channel is a non-DM channel, hide the online indicator
+        presenceAvatarView.isOnlineIndicatorVisible = false
+        
+        let lastActiveMembers = channel.lastActiveMembers
```

Comment:
this line is the copypase of line 86, please make a separate function for it and DRY it out

https://uk.wikipedia.org/wiki/Don%27t_repeat_yourself